controllers/topology/desired_state.go

```diff
@@ -66,16 +68,23 @@ func (r *ClusterReconciler) computeDesiredState(ctx context.Context, s *scope.Sc
 	// InfrastructureCluster and the ControlPlane objects generated by the previous step.
 	desiredState.Cluster = computeCluster(ctx, s, desiredState.InfrastructureCluster, desiredState.ControlPlane.Object)
 
-	// If required by the blueprint, compute the desired state of the MachineDeployment objects for the worker nodes, if any.
-	if !s.Blueprint.HasMachineDeployments() {
-		return desiredState, nil
+	// If required, compute the desired state of the MachineDeployments from the list of MachineDeploymentTopologies
+	// defined in the cluster.
+	if s.Blueprint.HasMachineDeployments() {
+		desiredState.MachineDeployments, err = computeMachineDeployments(ctx, s, desiredState.ControlPlane)
+		if err != nil {
+			return nil, err
```

Comment:
Do we want to wrap this error?

Reply:
I'll take a closer look

Reply:
Yup. I'll do the same for other errors above.